// clash-royale-api/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

require("dotenv/config");

var _Card = _interopRequireDefault(require("./Card"));

var _Clan = _interopRequireDefault(require("./Clan"));

var _Location = _interopRequireDefault(require("./Location"));

var _Player = _interopRequireDefault(require("./Player"));

var _Tournament = _interopRequireDefault(require("./Tournament"));

var _GlobalTournament = _interopRequireDefault(require("./GlobalTournament"));

var _utils = require("./utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class Client {
  constructor(token = null, request = _utils.requester) {
    this.token = token || process.env.CLASH_ROYALE_API_TOKEN;
    console.log("Test class: valid token" )
    if (!this.token) {
      throw new Error(`Invalid Token: ${token}`);
    }
    console.log("Before" )

    this.request = (0, _utils.requester)(this.token);
    console.log("After" )

  }

  // async cards() {
  //   const response = await this.request('cards');
  //   return new _Card.default(response);
  // }

  async clans(params) {
    let query = params;

    if (typeof params === 'string') {
      query = {
        name: params
      };
    }

    const response = await this.request('clans', query);
    return new _Clan.default(response);
  }

  async clan(tag, path) {
    const response = await this.request(`clans/${encodeURIComponent(tag)}/${path || ''}`);
    return new _Clan.default(response);
  }

  // async locations(params = {}) {
  //   const response = await this.request('locations', params);
  //   return new _Location.default(response);
  // }

  // async location(id, path, query) {
  //   const response = await this.request(`locations/${id}/${path || ''}`, query);
  //   return new _Location.default(response);
  // }

  // async player(tag, path) {
  //   const response = await this.request(`players/${encodeURIComponent(tag)}/${path || ''}`);
  //   return new _Player.default(response);
  // }

  // async tournaments(params) {
  //   let query = params;

  //   if (typeof params === 'string') {
  //     query = {
  //       name: params
  //     };
  //   }

  //   const response = await this.request('/tournaments', query);
  //   return new _Tournament.default(response);
  // }

  // async tournament(tag) {
  //   const response = await this.request(`/tournaments/${encodeURIComponent(tag)}`);
  //   return new _Tournament.default(response);
  // }

  // async globalTournaments() {
  //   const response = await this.request('globaltournaments');
  //   return new _GlobalTournament.default(response);
  // }

}

var _default = Client;
exports.default = _default;
module.exports = exports.default;
